middleware/http/server.go

```diff
@@ -135,10 +143,10 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		code := ri.getStatusCode()
 		sCode := strconv.Itoa(code)
 		if code > 399 {
-			zipkin.TagError.Set(sp, sCode)
+			h.errHandler(sp, nil, code)
 		}
 		zipkin.TagHTTPStatusCode.Set(sp, sCode)
-		if h.tagResponseSize && ri.size > 0 {
+		if h.tagResponseSize && atomic.LoadUint64(&ri.size) > 0 {
```

Comment:
Sometimes I get this:

![image](https://user-images.githubusercontent.com/3075074/57769790-11e5d300-770f-11e9-9a1a-8c7e8b987aa2.png)

This is the only fix I can think of.